crates/lang/codegen/src/generator/mod.rs

```diff
@@ -12,6 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/// Implements `AsRef` for a code generator type.
+///
+/// Code generators always have a shared `contract` reference to the contract.
+/// They need to implement this trait in order to use other code generators.
+macro_rules! impl_as_ref_for_generator {
```

Comment:
nit: I'd move this to after the imports

Reply:
then it won't work ... Rust's macro system needs this ordering unfortunately.

Reply:
I feel like I knew that deep down, but TIL